src/js/site/main.js

```diff
@@ -402,3 +402,72 @@ $(document).ready(function () {
 
 // Currently this function is only used for the insights table
 $(document).ready(highlightURLHash());
+
+// update date shown to be X ago
+$(document).ready(function () {
+  // tooltip code for posted on time
+  const tooltiptime = document.getElementById('tooltiptime');
+  let timeposted = document.getElementById('time-posted-on');
+  let popperInstance = null;
+  var showEvents = ['mouseenter', 'focus'];
+  var hideEvents = ['mouseleave', 'blur'];
+
+  showEvents.forEach((event) => {
+    timeposted.addEventListener(event, () => {
+      tooltiptime.setAttribute('data-show', '');
+      popperInstance = createPopper(timeposted, tooltiptime, {});
+    });
+  });
+
+  hideEvents.forEach((event) => {
+    timeposted.addEventListener(event, () => {
+      tooltiptime.removeAttribute('data-show');
+      if (popperInstance) {
+        popperInstance.destroy();
+        popperInstance = null;
+      }
+    });
+  });
+
+  // change date into how long ago that date is relative to when view page
+  function ago(date) {
+    function render(n, unit) {
+      return n + ' ' + unit + (n === 1 ? '' : 's') + ' ago';
+    }
+
+    var seconds = Math.floor((new Date() - date) / 1000);
+
+    var interval = Math.floor(seconds / (60 * 60 * 24 * 365));
+    if (interval >= 1) {
+      return '+1 year ago';
+    }
+    interval = Math.floor(seconds / (60 * 60 * 24 * 30));
+    if (interval >= 1) {
+      return render(interval, 'month');
+    }
+    interval = Math.floor(seconds / (60 * 60 * 24));
+    if (interval >= 1) {
+      return render(interval, 'day');
+    }
+    interval = Math.floor(seconds / (60 * 60));
+    if (interval >= 1) {
+      return render(interval, 'hour');
+    }
+    interval = Math.floor(seconds / 60);
+    if (interval >= 1) {
+      return render(interval, 'minute');
+    }
+    interval = Math.floor(seconds);
+    return render(interval, 'second');
+  }
```

Comment:
This seems like something we might want to move to our [utils](https://github.com/circleci/circleci-docs/blob/407b3e51565e647f55e707eaea2b2d1f05d85f8e/src/js/utils.js) file.